Makefile

```diff
@@ -78,8 +78,14 @@ tested-$(ISOLATION_ID): $(HOST_ARCHITECTURE)-ensure-context
 	rm -rf .artifacts
 	mkdir -p .artifacts
 
-	container_id=$$(docker create tested-artifacts:${ISOLATION_ID}); \
-		docker cp $$container_id:/artifacts `pwd`/.artifacts/ \
+	# "cargo test" itself uses Docker from within the container
```

Comment:
I have never tried this with buildx, but we should be able to docker-in-docker from cargo directly ?

Reply:
Not seeing any clear way to, alas.

Reply:
So you are putting everything together and then running the test outside of the image build?

Reply:
If we *can* get these tests running inside the image build as before then that would be great, it would probably solve my other to-do here of it running all the cargo downloading and compiling of libraries again even if I run a `cargo test` with `--no-run` inside the build first.